UChicago VuFind.js

```diff
@@ -0,0 +1,794 @@
+		return 'videoRecording';
+	} else if (doc.querySelector('.format.dissertations')) {
+		return 'thesis';
+	} else if (doc.querySelector('.format.audio')) {
+		return 'audioRecording';
+	} else if (doc.querySelector('.format.map')) {
+		return 'map';
+	} else {
+		return 'book';
+	}
+	// VuFind URL patterns starting with 'Search' are for search results
+	} else if (url.includes('Search')) {
+	return 'multiple';
+	// the translator should do nothing on every other URL pattern
+	} else {
+	return false;
+	}
+}
+
+const doWeb = (doc, url) => {
+	if (detectWeb(doc, url) == 'multiple') {
+	// ingest multiple MARC records
+	Zotero.selectItems(getSearchResults(doc), items => {
+		if (items) {
+		itemURLs = Object.keys(items);
+		itemURLs.map(scrapeMARC(doc));
+		}
+	});
+
+	} else {
+	// ingest single MARC record
+	scrapeMARC(doc)(url);
+	}
+}
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://catalog.lib.uchicago.edu/vufind/Record/6422153",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Eat, pray, love: one woman's search for everything across Italy, India, and Indonesia",
+				"creators": [
+					{
+						"firstName": "Elizabeth",
+						"lastName": "Gilbert",
+						"creatorType": "author"
+					}
+				],
+				"date": "2007",
+				"ISBN": "9780143038412",
+				"callNumber": "G154.5.G55 A3 2007",
+				"extra": "OCLC: 82462742",
+				"libraryCatalog": "UChicago VuFind",
+				"numPages": "334",
+				"place": "New York",
+				"publisher": "Penguin",
+				"shortTitle": "Eat, pray, love",
+				"url": "https://catalog.lib.uchicago.edu/vufind/Record/6422153",
+				"attachments": [],
+				"tags": [
+					{
+						"tag": "Biography"
+					},
+					{
+						"tag": "Biography"
+					},
+					{
+						"tag": "Gilbert, Elizabeth"
+					},
+					{
+						"tag": "Gilbert, Elizabeth"
+					},
+					{
+						"tag": "Travel"
+					},
+					{
+						"tag": "Travel"
+					},
+					{
+						"tag": "Travel writers"
+					},
+					{
+						"tag": "Travel writers"
+					},
+					{
+						"tag": "United States"
+					},
+					{
+						"tag": "United States"
+					}
+				],
+				"notes": [
+					{
+						"note": "Previously published: New York : Viking Penguin, 2006"
+					},
+					{
+						"note": "Italy, or, \"Say it like you eat it,\" or, 36 tales about the pursuit of pleasure -- India, or, \"Congratulations to meet you,\" or, 36 tales about the pursuit of devotion -- Indonesia, or, \"Even in my underpants, I feel different,\" or, 36 tales about the pursuit of balance"
+					}
+				],
+				"seeAlso": []
+			}
+		]
+	},
+	{
+		"type": "web",
+		"url": "https://catalog.lib.uchicago.edu/vufind/Record/10773190",
+		"items": [
+			{
+				"itemType": "thesis",
+				"title": "Characterizing kinds: A semantics for generic sentences",
+				"creators": [
+					{
+						"firstName": "Matthew",
+						"lastName": "Teichman",
+						"creatorType": "author"
+					},
+					{
+						"lastName": "University of Chicago",
```

Comment:
It looks like this should be in the `university` field and the current value in the `university field` (ProQuest) should be... maybe nowhere? Maybe in `archive`?

If this is a common issue, can we correct it automatically?

Reply:
So the rest of these changes, the linting, etc. are pretty straightforward, I can get to work on those right away.  This one is really interesting but it is also more of a rabbit hole, because it involves the MARC import translator.  I think I need to ask you a couple questions, and then you'll have a chance to look at them while I make the rest of your suggested changes.

Phenomenon No. 1: "University of Chicago" being listed as a secondary creator ("contributor") in addition to me being listed as the author.  This is definitely weird!  I believe it is the MARC import translator code that is making this happen.  One initial indication is that if you comment out line 155, the same contributor with the "last name" of University of Chicago continues to appear in the entry.  You can observe phenomenon No. 1 happening in other dissertations in our collection as well:

https://catalog.lib.uchicago.edu/vufind/Record/10168855
https://catalog.lib.uchicago.edu/vufind/Record/10773211
https://catalog.lib.uchicago.edu/vufind/Record/11715209

It looks like the MARC import translator is pulling "University of Chicago" from the 710|a MARC fields and adding it as a second author with 710|a as its "last name" and classified as a "contributor".  See here:

https://github.com/zotero/translators/blob/master/MARC.js#L409

Assuming that's right, how would you prefer to handle this?

Reply:
I'll let @AbeJellinek comment on the specifics, but just to clarify, Publisher and University are the same thing — the field is [just called University when it's a thesis](https://github.com/zotero/zotero-schema/blob/628358f04b967a7669063cc3713d6fe593750570/schema.json#L2257-L2271).

Reply:
@dstillman thanks for the link to the schema; that explains the behavior perfectly!

Reply:
All the changes except a) the uncurrying of `scrapeMARC` and b) whatever we're going to do about the university as a contributor and ProQuest not being a university are now up on the branch.  Please feel free to ping me if I can help with other stuff or answer any other questions.

Reply:
How should these actually be cited? Is ProQuest as publisher (university) correct? If so, I think the way it works now is fine. If not, we probably need to fix the MARC translator.

Reply:
No, ProQuest should definitely not be the University (nor CSL publisher). Here's the sample entry from bwiernik's APA library, I think following this (i.e. putting ProQuest into archive) makes sense: https://www.zotero.org/groups/2205533/items/UD3VZHS5

Reply:
So since you brought this to our attention, our metadata analysts have been discussing the issue and think that this could be something to fix at the catalog level.  That is, for recent dissertations, ProQuest should not be listed as the publisher, as is the norm right now.  Actually doing the fix is somewhat logistically complicated, but they are looking into it and apparently there's interest.

In terms of this pull request, it's a little unclear to me what the fix would be at the level of the VuFind web translator code, but one thing I could at least do is replace that test case with a test case involving an older dissertation that lists the actual university as publisher.  For example, this one:

https://catalog.lib.uchicago.edu/vufind/Record/7143200

What do you think, would that be suitable from your point of view?

Reply:
That would be fine, I think. Something like

> - check if the 710|e field is "degree granting institution"
> - if it is, change item.publisher to whatever's in the 710|a field
> - if not, change item.publisher to the empty string, and then nothing ends up populating the "university" field (or maybe you have a bibliographic field delete method somewhere in your project that could be used for that)

Also sounds fine for the time being, if that 710|e value is consistent.

Reply:
(With @adam3smith's suggestion to put ProQuest in `archive`.)

Reply:
Ok, these changes are now in.